proto/spire/api/server/localauthority/v1/localauthority.proto

```diff
@@ -0,0 +1,189 @@
+        // has been prepared but not activated yet), a FailedPrecondition
+        // error will be returned.
+        rpc RevokeJWTAuthority(RevokeJWTAuthorityRequest) returns (RevokeJWTAuthorityResponse);
+
+        // GetX509AuthorityState returns the state of all locally configured
+        // X.509 authorities.
+        rpc GetX509AuthorityState(GetX509AuthorityStateRequest) returns (GetX509AuthorityStateResponse);
+
+        // PrepareX509Authority prepares a new X.509 authority for use by
+        // generating a new key and injecting the resulting CA certificate into
+        // the bundle. This action will  propagate the new CA cluster-wide.
+        rpc PrepareX509Authority(PrepareX509AuthorityRequest) returns (PrepareX509AuthorityResponse);
+
+        // ActivateX509Authority activates a prepared X.509 authority for use,
+        // which will cause it to be used for all X.509 signing operations
+        // serviced by this server going forward. If a new X.509 authority has
+        // not already been prepared, a FailedPrecondition error will be returned.
+        rpc ActivateX509Authority(ActivateX509AuthorityRequest) returns (ActivateX509AuthorityResponse);
+
+        // TaintX509Authority marks the previously active X.509 authority as
+        // being tainted. SPIRE Agents observing an authority to be tainted
+        // will perform proactive rotations of any key material related to
+        // the tainted authority. The result of this action will be observed
+        // cluster-wide.
+        // It can receive the public key of an old X.509 authority.
+        //
+        // If a previously active X.509 authority does not exist (e.g. if one
+        // has been prepared but not activated yet), a FailedPrecondition
+        // error will be returned.
+        rpc TaintX509Authority(TaintX509AuthorityRequest) returns (TaintX509AuthorityResponse);
+
+        // RevokeX509Authority revokes the previously active X.509 authority by
+        // removing it from the bundle and propagating this update throughout
+        // the cluster.
+        // It can receive the public key of an old X.509 authority.
+        //
+        // If a previously active X.509 authority does not exist (e.g. if one
+        // has been prepared but not activated yet), a FailedPrecondition
+        // error will be returned.
+        rpc RevokeX509Authority(RevokeX509AuthorityRequest) returns (RevokeX509AuthorityResponse);
+}
+
+message GetJWTAuthorityStateRequest {}
+
+message GetJWTAuthorityStateResponse {
+        repeated AuthorityState states = 1;
+}
+
+message PrepareJWTAuthorityRequest {}
+
+message PrepareJWTAuthorityResponse {
+        AuthorityState prepared_authority = 1;
+}
+
+message ActivateJWTAuthorityRequest {}
+
+message ActivateJWTAuthorityResponse {
+        AuthorityState activated_authority = 1;
+}
+
+message TaintJWTAuthorityRequest {
+        // The key ID of the local authority to taint
+        string key_id = 1;
```

Comment:
the idea is to use "old"  authority if no key_id is provided, and X.509 implementation requires a public key... 
I'm worried about the difference about both, we may use public key here too..
but our endpoints returns a key_id so I think this can be more natural to use that key id (and our JWT SVIDs has the key ID too)

Reply:
ok, so move to taint using public keys on both looks like the correct call.